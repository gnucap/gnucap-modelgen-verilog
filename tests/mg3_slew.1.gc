
attach ../modelgen_0.so


verilog
// ddx test case inspired by LRM 4.5.6

`modelgen
module test_slw1(p,c);
	inout p, c;
	electrical p, c;
	parameter real gm = 1.0;
	(* desc="in" *) real vin;
	analog begin
		vin = V(c);
		V(p) <+ slew(vin, 1, -1);
	end
endmodule

!make test_slw1.so > /dev/null
attach ./test_slw1.so

test_slw1 #() dut(2,1);

spice

Vin 1 0 sin f=1 a=1 delay={-.1}
.print tran v(nodes)
.status notime
.tran 1
