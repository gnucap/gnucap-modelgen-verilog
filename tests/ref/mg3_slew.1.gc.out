#Time       v(1)       v(2)      
 0.         0.58779    0.58779   
 0.01       0.63742    0.59779   
 0.02       0.68455    0.60779   
 0.062609   0.853      0.65039   
 0.15635    0.9992     0.74413   
 0.29696    0.60315    0.60352   
 0.43757   -0.23384    0.46292   
 0.57817   -0.89988    0.32231   
 0.71878   -0.90806    0.1817    
 0.85939   -0.25239    0.04109   
 0.91122    0.07041    0.07041   
 1.         0.58779    0.15919   
